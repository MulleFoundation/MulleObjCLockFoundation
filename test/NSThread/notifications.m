#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
# pragma message "this test does not work with Apple Foundation"
#else
# import <MulleObjC/MulleObjC.h>
#endif


@interface NSThread( Print)
@end


@implementation NSThread( Print)


- (void) _isProbablyGoingSingleThreaded
{
   printf( "%s\n", __PRETTY_FUNCTION__);
   fflush( stdout);
}


- (void) _isGoingMultiThreaded
{
   printf( "%s\n", __PRETTY_FUNCTION__);
   fflush( stdout);
}


- (void) _threadWillExit
{
   printf( "%s\n", __PRETTY_FUNCTION__);
   fflush( stdout);
}


+ (void) doNothing:(id) obj
{
   printf( "%s\n", __PRETTY_FUNCTION__);
   fflush( stdout);
}


+ (void) sleep1:(id) obj
{
   sleep( 1);
   printf( "%s\n", __PRETTY_FUNCTION__);
   fflush( stdout);
}

@end



int main( void)
{
   printf( "START: %s\n", __FUNCTION__);

   [NSThread detachNewThreadSelector:@selector( doNothing:)
                            toTarget:[NSThread class]
                          withObject:nil];
   [NSThread detachNewThreadSelector:@selector( sleep1:)
                            toTarget:[NSThread class]
                          withObject:nil];
   sleep( 2);
   printf( "STOP: %s\n", __FUNCTION__);
   return( 0);
}
