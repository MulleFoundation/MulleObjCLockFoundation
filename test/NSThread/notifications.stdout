START: main
-[NSThread(Print) _isGoingMultiThreaded]
+[NSThread(Print) doNothing:]
-[NSThread(Print) _threadWillExit]
+[NSThread(Print) sleep1:]
-[NSThread(Print) _threadWillExit]
-[NSThread(Print) _isProbablyGoingSingleThreaded]
STOP: main
